google/datalab/bigquery/commands/_bigquery.py

```diff
@@ -316,6 +316,73 @@ def _create_load_subparser(parser):
   return load_parser
 
 
+def _create_pipeline_subparser(parser):
+  pipeline_parser = parser.subcommand('pipeline', 'Load from GCS, query, and extract data from GCS '
+                                                  'into a BigQuery table. If creating a new table, '
+                                                  'a schema should be specified in YAML or JSON in '
+                                                  'the cell body, otherwise the schema is inferred '
+                                                  'from existing table.')
+
+  # common arguments
+  pipeline_parser.add_argument('-b', '--billing', type=int, help='BigQuery billing tier')
+  # TODO(rajivpb): I've naively made this a common param. Any objections?
+  pipeline_parser.add_argument('-nc', '--nocache', help='Don\'t use previously cached results',
+                               action='store_true')
+
+  # load arguments
+  pipeline_parser.add_argument(
+      '-lm', '--load_mode', help='One of create (default), append or overwrite for loading data',
+      choices=['create', 'append', 'overwrite'], default='create')
+  pipeline_parser.add_argument('-lf', '--load_format', help='The format of the source file in GCS',
+                               choices=['json', 'csv'], default='csv')
+  pipeline_parser.add_argument('--skip',
+                               help='The number of head lines to skip during load; useful for CSV',
+                               type=int, default=0)
+  pipeline_parser.add_argument('-s', '--strict', help='Whether to reject bad values and jagged '
+                                                      'lines when loading', action='store_true')
+  pipeline_parser.add_argument(
+      '-ld', '--load_delimiter', default=',', help='The inter-field delimiter for CVS (default ,) '
+                                                   'in load file')
+  pipeline_parser.add_argument(
+      '-lq', '--load_quote', default='"', help='The quoted field delimiter for CVS (default ") in '
+                                               'the load file')
+  pipeline_parser.add_argument('-lp', '--load_path', help='The path URL of the GCS load file(s)')
+  pipeline_parser.add_argument('-lt', '--load_table', help='The destination bigquery table name '
+                                                           'for loading.')
+
+  # execute arguments
+  pipeline_parser.add_argument(
+      '-em', '--execute_mode', help='The table creation mode for execution', default='create',
+      choices=['create', 'append', 'overwrite'])
+  pipeline_parser.add_argument(
+      '-l', '--large', help='Whether to allow large results during execution', action='store_true')
+  pipeline_parser.add_argument('-eq', '--execute_query', help='The name of query for execution',
+                               required=True)
+  pipeline_parser.add_argument(
+      '-et', '--extract_table', help='Destination table name for the execution results')
```

Comment:
Similar to input, we should have an output section in the YAML that describes the output -- table or gcs, with similar support for placeholders to be able to generate per pipeline run outputs.

Reply:
Yeah, see response above for your comment on L334.